target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8070-eap620hd-v1.dts

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR BSD-2-Clause
+
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include "ipq8074-ess.dtsi"
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "TP-Link EAP620 HD v1";
+	compatible = "tplink,eap620hd-v1", "qcom,ipq8074";
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		led-boot = &led_status_blue;
+		led-failsafe = &led_status_blue;
+		led-running = &led_status_blue;
+		led-upgrade = &led_status_blue;
+	};
+
+	chosen {
+		stdout-path = "serial0,115200n8";
+		bootargs-append = " root=/dev/ubiblock0_1";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 50 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_blue: status-blue {
+			function = LED_FUNCTION_STATUS;
+			gpios = <&tlmm 42 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_BLUE>;
+		};
+	};
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+};
+
+&mdio {
+	status = "okay";
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 37 GPIO_ACTIVE_LOW>;
+	
+	ethernet-phy-package@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "qcom,qca8075-package";
+		reg = <0>;
+
+		qca8075_4: ethernet-phy@4 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <4>;
+		};
+	};
+};
+
+&dp6 {
+	status = "okay";
+	phy-handle = <&qca8075_4>;
+	label = "lan";
+};
+
+&switch {
+	status = "okay";
+	switch_lan_bmp = <ESS_PORT6>;
+	switch_mac_mode0 = <MAC_MODE_PSGMII>;
+	switch_mac_mode1 = <MAC_MODE_SGMII_CHANNEL0>;
+	switch_mac_mode2 = <MAC_MODE_SGMII_CHANNEL0>;
+
+	qcom,port_phyinfo {
+		port@6 {
+			phy_address = <4>;
+			port_id = <6>;
```

Comment:
This looks weird, since I am not sure if QCA807x even does regular SGMII mode at all.

Reply:
I just went off what the stock firmware's device tree had, I can experiment with changing it if you thing this is incorrect

Reply:
Well, the thing is that I dont see how this could actually work, especially since according to this its connected to UNIPHY2 via SGMII?

I have not seen QCA807x connected to UNIPHY1 and UNIPHY2 as they dont support QSGMII nor PSGMII which QCA807x uses.

Can you install `mdio-tools` and give me the output of:
`mdio 9* 0x5 0x1f`

Cause the only way I see this working is in a QSGMII+SGMII config where they are only using the SGMII link for the last port basically.

Reply:
```
root@OpenWrt:~# mdio 9*
 DEV      PHY-ID  LINK
0x04  0x004dd074  up
root@OpenWrt:~# mdio 9* 0x4
BMCR(0x00): 0x3100
  flags: -reset -loopback +aneg-enable -power-down -isolate -aneg-restart
         -collision-test
  speed: 100-full

BMSR(0x01): 0x796d
  capabilities: -100-t4 +100-tx-f +100-tx-h +10-t-f +10-t-h -100-t2-f -100-t2-h
  flags:        +ext-status +aneg-complete -remote-fault +aneg-capable +link
                -jabber +ext-register

ID(0x02/0x03): 0x004dd074

ESTATUS(0x0F): 0xa000
  capabilities: +1000-x-f -1000-x-h +1000-t-f -1000-t-h
root@OpenWrt:~# mdio 9* 0x4 0x1f
0x8111
root@OpenWrt:~# mdio 9* 0x5 0x1f
0xffff
root@OpenWrt:~#

Reply:
Well, that is AR8031/3 PHY, that makes a lot more sense as that is an SGMII or RGMII PHY.

In that case, the whole qca807x package node is incorrect.

Reply:
Oops, my mistake